packages/compiler/src/render3/r3_template_transform.ts

```diff
@@ -51,33 +51,46 @@ export interface Render3ParseResult {
   styles: string[];
   styleUrls: string[];
   ngContentSelectors: string[];
+  // Will be defined if `Render3ParseOptions['collectCommentNodes']` is true
+  comments?: t.Comment[];
+}
+
+interface Render3ParseOptions {
+  collectCommentNodes: boolean;
 }
 
 export function htmlAstToRender3Ast(
-    htmlNodes: html.Node[], bindingParser: BindingParser): Render3ParseResult {
-  const transformer = new HtmlAstToIvyAst(bindingParser);
+    htmlNodes: html.Node[], bindingParser: BindingParser,
+    options: Render3ParseOptions): Render3ParseResult {
+  const transformer = new HtmlAstToIvyAst(bindingParser, options);
   const ivyNodes = html.visitAll(transformer, htmlNodes);
 
   // Errors might originate in either the binding parser or the html to ivy transformer
   const allErrors = bindingParser.errors.concat(transformer.errors);
 
-  return {
+  const result: Render3ParseResult = {
     nodes: ivyNodes,
     errors: allErrors,
     styleUrls: transformer.styleUrls,
     styles: transformer.styles,
-    ngContentSelectors: transformer.ngContentSelectors,
+    ngContentSelectors: transformer.ngContentSelectors
   };
+  if (options.collectCommentNodes) {
+    result.comments = transformer.comments;
```

Comment:
Just a small comment/proposal: may be we cab use `commentNodes` as a field name (here and in other related places)? That would correspond to the `collectCommentNodes` config option better and also provide more context even without looking at types (i.e. it'd be more obvious what these fields contain/represent).
```suggestion
    result.commentNodes = transformer.commentNodes;
```

Reply:
Done